jdbc-perf-logger-driver/src/main/java/ch/sla/jdbcperflogger/logger/PerfLogger.java

```diff
@@ -275,4 +276,45 @@ static String getValueAsString(final SqlTypedValue sqlTypedValue, final Database
         strBuilder.append("*/");
         return strBuilder.toString();
     }
+    public static String removeComments(String result){
+        if(result == null || "".equals(result) || result.length() < 4)
+            return result;
+        //do a try catch the comments are never closed or empty string, should not happen,
+        try {
+            boolean inString = false;
+            StringBuffer cleanedStatement = new StringBuffer();
+            for (int i = 0; i < result.length(); i++) {
+                //take care of the escape char
+                if (result.charAt(i) == '\'') {
+                    inString = !inString;
+                }
+                if (!inString) {
+                    if (isStartingComment(result, i)) {
+                        i = i + 2;
+                        while (!isEndComment(result, i)) {
+                            i++;
+                        }
+                        i = i + 2;
+                    }
+                }
+                if (i < result.length()) {
+                    cleanedStatement.append(result.charAt(i));
+
+                }
+            }
+            if(inString) return result;
+
+            return cleanedStatement.toString();
+        }
+        catch (StringIndexOutOfBoundsException e) {
```

Comment:
agree with you. 
The point as you noticed is to avoid to check each time the out of index and return a value which will be interpreted as a malformed statement and then return the original statement.